src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -1146,7 +1147,11 @@ private PreparedStatement getUpdateStreamFromDb(Connection conn)
       try (Connection conn = makeNewConnection()) {
         for (DocId id : ids) {
           log.log(Level.FINE, "about to get acl of doc {0}", id);
-          Acl acl = getAcl(conn, id.getUniqueId());
+          Acl acl = getAcl(conn, null, id.getUniqueId());
```

Comment:
You missed some changes here. Try running this code live by fetching public docs directly from the adaptor.